src/test/java/io/lettuce/core/cluster/pubsub/RedisClusterPubSubConnectionIntegrationTests.java

```diff
@@ -100,33 +96,25 @@ void testRegularClientPubSubChannels() {
         assertThat(channelsOnOtherNode).isEmpty();
     }
 
-    @Test
-    void testRegularClientPubSubShardChannels() {
-
-        String nodeId = pubSubConnection.sync().clusterMyId();
-        RedisClusterNode otherNode = getOtherThan(nodeId);
-        /// TODO : uncomment after SSUBSCRIBE is implemented
-        // pubSubConnection.sync().ssubscribe(key);
-
-        List<String> channelsOnSubscribedNode = connection.getConnection(nodeId).sync().pubsubShardChannels();
-        // assertThat(channelsOnSubscribedNode).hasSize(1);
-
-        List<String> channelsOnOtherNode = connection.getConnection(otherNode.getNodeId()).sync().pubsubShardChannels();
-        assertThat(channelsOnOtherNode).isEmpty();
-    }
-
     @Test
     @EnabledOnCommand("SSUBSCRIBE")
     void subscribeToShardChannel() throws Exception {
-        pubSubConnection.addListener(connectionListener);
-        pubSubConnection.async().ssubscribe(shardChannel);
-        assertThat(connectionListener.getChannels().poll(3, TimeUnit.SECONDS)).isEqualTo(shardChannel);
+        pubSubConnection.sync().ssubscribe(shardChannel);
+
+        Wait.untilEquals(1L, connectionListener.getShardCounts()::poll).waitOrTimeout();
+        Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
     }
 
+    @Disabled
+    // This test is currently failing because the replica of the master node, where we subscribe to a shard channel,
+    // could be used to SPUBLISH to this channel, but does not list the shard channels with PUBSUB SHARDCHANNELS or
+    // PUBSUB SHARDNUMSUB
+
+    // furthermore the test does not address the possibility that the SSUBSCRIBE could result in a MOVED, e.g. when
+    // the hash of the shard channel name falls into the slot space of another node
     @Test
-    @EnabledOnCommand("SSUBSCRIBE")
+    // @EnabledOnCommand("SSUBSCRIBE")
     void subscribeToShardChannelViaOtherEndpoint() throws Exception {
-        pubSubConnection.addListener(connectionListener);
```

Comment:
please use `shardTestChannel` instead of `shardChannel` and it should run fine. 
use of `shardChannel` must be a typo or copy/paste issue.

Reply:
Okay the original issue with this test is that it never called `get()` on the async command and thus it was never executed. When we do that we get a MOVED response.

This contradicts the documentation about [clustered connections](https://lettuce.io/core/release/reference/#redis-cluster) so I will explore further.